lesson10/lesson10.js

```diff
@@ -0,0 +1,57 @@
+const ul = document.getElementById('js-list');
+const listItems = [
+  { to: "bookmark.html", img: "bookmark-icon.png", alt: "画像1", text: "ブックマーク" },
+  { to: "message.html", img: "mail-icon.png", alt: "画像2", text: "メッセージ" }
+];
+
+function showLoading() {
+  const loadingImg = document.createElement('img');
+  loadingImg.src = "loading-circle.gif";
+  loadingImg.id = "loading-img";
+  ul.appendChild(loadingImg);
+}
+
+function removeLoading() {
+  const loadingID = document.getElementById('loading-img');
+  loadingID.remove();
+}
+
+function createList(item) {
+  const fragment = document.createDocumentFragment();
+  for (let i = 0; i <= item.length - 1; i++) {
+    const li = document.createElement('li');
+    const a = document.createElement('a');
+    const img = document.createElement('img');
+    a.textContent = item[i].text;
+    a.href = item[i].to;
+    img.src = item[i].img;
+    img.alt = item[i].alt;
+    fragment.appendChild(li).appendChild(a).appendChild(img);
+  }
+  ul.appendChild(fragment)
+}
+
+function fetchListData() {
+  const promise = new Promise((resolve) => {
+    setTimeout(() => resolve(listItems), 3000)
+  })
+  return promise;
+}
+
+async function asyncProcessing() {
+  showLoading();
+  const item = await fetchListData();
+  removeLoading();
+  createList(item);
+}
```

Comment:
> After getting the data in the list, execute asyncProcessing () ;.

At present, fetchListData has been executed twice.
- In the try
- In asyncProcessing()

How about taking the return value of the fetchData() executed in the try and executing it here?

Reply:
Thank you for your suggestion! I tried it. We apologize for the inconvenience, but we would appreciate it if you could confirm it.
https://github.com/rikokko/to_be_frontend/pull/18/commits/fe28269c60862c292f82644a4ddc7a42f1398279